test/priority_queue_test.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include <openssl/opensslconf.h>
+#include <internal/priority_queue.h>
+#include <openssl/err.h>
+#include <openssl/crypto.h>
+
+#include "internal/nelem.h"
+#include "testutil.h"
+
+/* The macros below generate unused functions which error out one of the clang
+ * builds.  We disable this check here.
```

Comment:
This seems like a problem for when we start using this. Should the unused functions be marked with `ossl_unused`?

Reply:
Hmmm, it works without this.  So removed.